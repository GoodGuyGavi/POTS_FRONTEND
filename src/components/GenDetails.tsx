import React from "react";
import { Card, ListGroup, ListGroupItem } from "react-bootstrap";
import ButtonApp from "./Button";

export interface IGenDetailsProps {
  list?: any;
  title?: any;
  buttonTitle?: any;
}

const GenDetails = (props: IGenDetailsProps) => {
  const { title, buttonTitle } = props;
  return (
    <Card style={{ width: "18rem" }}>
      <Card.Body>
        <Card.Title>{title}</Card.Title>
      </Card.Body>
      <ListGroup className="list-group-flush">
        {props.list.map((d: any) => (
          <ListGroupItem>{d.detail}</ListGroupItem>
        ))}
      </ListGroup>
      <Card.Body>
        <ButtonApp buttonTitle={buttonTitle} variant="secondary" />
      </Card.Body>
    </Card>
  );
};

export default GenDetails;
