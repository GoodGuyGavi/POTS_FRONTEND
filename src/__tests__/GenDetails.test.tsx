/**
 * @format
 */

import React from "react";
import { shallow } from "enzyme";
import GenDetails from "../components/GenDetails";

const genDetailList = [
  {
    key: 1,
    detail: "This is first details"
  },
  {
    key: 2,
    detail: "This is second details"
  }
];

describe("General Details", () => {
  test("renders correctly", () => {
    expect(
      shallow(
        <GenDetails
          buttonTitle="Modify Supplier Detail"
          list={genDetailList}
          title="General Details"
        />
      )
    ).toMatchSnapshot();
  });
});
