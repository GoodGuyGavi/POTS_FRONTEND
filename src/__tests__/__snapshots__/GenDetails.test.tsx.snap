// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General Details renders correctly 1`] = `ShallowWrapper {}`;
